sdk/openai/ai-openai/review/ai-openai.api.md

```diff
@@ -0,0 +1,194 @@
+## API Report File for "@azure/ai-openai"
+
+> Do not edit this file. It is a report generated by [API Extractor](https://api-extractor.com/).
+
+```ts
+
+import { AzureKeyCredential } from '@azure/core-auth';
+import { ClientOptions } from '@azure-rest/core-client';
+import { KeyCredential } from '@azure/core-auth';
+import { RawHttpHeadersInput } from '@azure/core-rest-pipeline';
+import { TokenCredential } from '@azure/core-auth';
+
+export { AzureKeyCredential }
+
+// @public
+export interface ChatChoice {
+    delta?: ChatMessage;
+    finishReason: CompletionsFinishReason | null;
+    index: number;
+    message?: ChatMessage;
+}
+
+// @public
+export interface ChatCompletions {
+    choices: ChatChoice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public (undocumented)
+export interface ChatCompletions {
+    choices: ChatChoice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public
+export interface ChatMessage {
+    content?: string;
+    role: ChatRole;
+}
+
+// @public
+export type ChatRole = string;
+
+// @public
+export interface Choice {
+    finishReason: CompletionsFinishReason | null;
+    index: number;
+    logprobs: CompletionsLogProbabilityModel | null;
+    text: string;
+}
+
+// @public
+export interface Completions {
+    choices: Choice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public (undocumented)
+export interface Completions {
+    choices: Choice[];
+    created: number;
+    id: string;
+    usage: CompletionsUsage;
+}
+
+// @public
+export type CompletionsFinishReason = string;
+
+// @public
+export interface CompletionsLogProbabilityModel {
+    textOffset: number[];
+    tokenLogprobs: (number | null)[];
+    tokens: string[];
+    topLogprobs: Record<string, number | null>[];
+}
+
+// @public
+export interface CompletionsLogProbabilityModel {
+    textOffset: number[];
+    tokenLogprobs: (number | null)[];
+    tokens: string[];
+    topLogprobs: Record<string, number | null>[];
+}
+
+// @public
+export interface CompletionsUsage {
+    completionTokens: number;
+    promptTokens: number;
+    totalTokens: number;
+}
+
+// @public
+export interface EmbeddingItem {
+    embedding: number[];
+    index: number;
+}
+
+// @public
+export interface Embeddings {
+    data: EmbeddingItem[];
+    usage: EmbeddingsUsage;
+}
+
+// @public (undocumented)
+export interface Embeddings {
+    data: EmbeddingItem[];
+    usage: EmbeddingsUsage;
+}
```

Comment:
@joheredi could code gen recognize whether there is a need to generate a model? these two are identical because the names in them are just one word.

Reply:
Removed manually in https://github.com/Azure/azure-sdk-for-js/pull/25921/commits/94342f75a9301a61d03bb0a57dcee8bc50d79b98